crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -860,7 +861,73 @@ where
         Ok((related_sid, span_relation))
     }
 
-    async fn end_invocation(
+    async fn end_invocation<State: StateReader + ReadOnlyJournalTable>(
+        &mut self,
+        state: &mut State,
+        effects: &mut Effects,
+        full_invocation_id: FullInvocationId,
+        invocation_metadata: InvocationMetadata,
+    ) -> Result<(), Error> {
+        // Send response back, if anyone is interested
+        if !invocation_metadata.response_sinks.is_empty() {
+            let invocation_id = InvocationId::from(&full_invocation_id);
+
+            // Find last output entry
+            let mut output_entry = None;
+            for i in (0..invocation_metadata.journal_metadata.length).rev() {
+                if let JournalEntry::Entry(e) = state
+                    .get_journal_entry(&invocation_id, i)
+                    .await?
+                    .unwrap_or_else(|| panic!("There should be a journal entry at index {}", i))
+                {
+                    if e.ty() == EntryType::Output {
+                        output_entry = Some(e);
+                        break;
+                    }
+                }
+            }
+
+            if let Some(output_entry) = output_entry {
+                let_assert!(
+                    Entry::Output(OutputEntry { result }) =
+                        output_entry.deserialize_entry_ref::<Codec>()?
+                );
+                self.send_response_to_sinks(
+                    effects,
+                    &InvocationId::from(&full_invocation_id),
+                    invocation_metadata.response_sinks.clone(),
+                    result,
+                );
+            } else {
+                // We don't panic on this, although it indicates a bug at the moment.
+                warn!("Invocation completed without an output entry. This is not supported yet.");
+                self.send_response_to_sinks(
+                    effects,
+                    &InvocationId::from(&full_invocation_id),
+                    invocation_metadata.response_sinks.clone(),
+                    UNKNOWN_INVOCATION_ERROR,
+                );
```

Comment:
Wondering whether this is the right course of action in case of a missing output entry. The alternative could be to keep the invocation until the user intervenes (e.g. by killing it or updating the version so that one can make progress again). With this implementation, the invocation will be gone and the next invocation will continue.

Reply:
> Wondering whether this is the right course of action in case of a missing output entry. The alternative could be to keep the invocation until the user intervenes (e.g. by killing it or updating the version so that one can make progress again). With this implementation, the invocation will be gone and the next invocation will continue.

Ok will change that. I think we'll touch this again if we ever get to implement #899